src/docfx/lib/json/JsonUtility.cs

```diff
@@ -62,62 +62,77 @@ static JsonUtility()
         /// <summary>
         /// Fast pass to read MIME from $schema attribute.
         /// </summary>
-        public static string ReadMime(TextReader reader)
+        public static (int line, int column, string mime) ReadMime(TextReader reader)
         {
-            var schema = ReadSchema(reader);
+            var (line, column, schema) = ReadSchema(reader);
             if (schema is null)
-                return null;
+                return (line, column, null);
 
             // TODO: be more strict
             var mime = schema.Split('/').LastOrDefault();
             if (mime != null)
-                return Path.GetFileNameWithoutExtension(schema);
+                return (line, column, Path.GetFileNameWithoutExtension(schema));
 
-            return null;
+            return (line, column, null);
         }
 
         /// <summary>
         /// Fast pass to read the value of $schema specified in JSON.
         /// $schema must be the first attribute in the root object.
-        /// Assume input is a valid JSON. Bad input will be process though Json.NET
+        /// Assume input is a valid JSON. Bad input will be processed through Json.NET
         /// </summary>
-        public static string ReadSchema(TextReader reader)
+        public static (int line, int column, string schema) ReadSchema(TextReader reader)
         {
```

Comment:
If I had a second change to write this method, I would use `JsonTextReader`, that way it is simple and easy to get $schema and also line info.